migrations/versions/0314_populate_email_access.py

```diff
@@ -0,0 +1,43 @@
+"""
+
+Revision ID: 0314_populate_email_access
+Revises: 0313_email_access_validated_at
+Create Date: 2020-01-31 10:35:44.524606
+
+"""
+from alembic import op
+
+
+revision = '0314_populate_email_access'
+down_revision = '0313_email_access_validated_at'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    # if user has email_auth, set email_access_validated_at on last login, else set it at user created_at date.
+    op.execute("""
+        UPDATE
+            users
+        SET
+            email_access_validated_at = created_at
+        WHERE
+            email_access_validated_at IS NULL
+    """)
+    op.execute("""
+        UPDATE
+            users
+        SET
+            email_access_validated_at = logged_in_at
+        WHERE
+            auth_type = 'email_auth'
+        AND
+            logged_in_at IS NOT NULL
```

Comment:
do we need `AND email_access_validated_at IS NULL` in this query too?

Reply:
The function it serves is: if someone has sms_auth and they have just reset their password using email link, they now have value in the `email_access_validated_at` field.
That check ensures this is not overwritten by an older date.

It's not essential, but it does something :)

Reply:
ah fair enough - is it right then that we don't expect `email_access_validated_at` to be useful for users with email_auth because we just refer to `logged_in_at` instead?

or should `email_access_validated_at` be equal to `logged_in_at` for email auth?

Reply:
> or should email_access_validated_at be equal to logged_in_at for email auth?
Exactly this, `email_access_validated_at` is updated each time they log in for people with `email_auth`